index.html

```diff
@@ -0,0 +1,261 @@
+<html>
+<head>
+  <style>
+  table {
+      border-collapse: collapse;
+      table-layout: fixed;
+  }
+  
+  table, th, tr, td{
+      border: 1px solid black;
+      height: 30px;
+  }
+  
+  table, th{
+      width: 100%;
+      font-size: 25px;
+  }
+  
+  th, a{
+     cursor: default; 
+  }
+  
+  .inputField {
+      width: 100%;
+      padding: 0px;
+      border-width: 0;
+      height: 100%;
+      font-size: 17px;
+  }
+  
+  .btn{
+      text-align: center;
+      vertical-align: middle;
+      cursor: default;
+  }
+  
+  .center {
+      margin-bottom: 8px;
+  }
+  
+  .checkbox{
+      cursor: default;
+  }
+  
+  [name=DateCell], [name=DeleteCell]  {
+      font-size: initial;
+  }
+  
+  #addRowBtn {
+      height: 25px;
+      font: icon;
+  }
+  </style>
+</head>
+
+<body onload='myTable()'>
+<script>
+  function myTable() {
+      'use strict';
+      
+      var tbl = [],
+          sortASC = [true, true, true, true, true],
+          addRowBtn = document.getElementById('addRowBtn'),
+          headers = document.getElementsByClassName('label'),
+          colTypeToNum = {String:'0', Number:'1', Boolean:'2', Date:'3', Delete:'4'},
+          arr = Array.prototype.slice.call( headers );
+      
+      arr.forEach(function(item, index) {
+          item.addEventListener('click', sortByColumn);
+      });
+      
+      addRowBtn.addEventListener('click', function addRow() {
+          var table = document.querySelector("#tblBody"),
+              row = table.insertRow(-1),
+              nameCell = row.insertCell(0),
+              numberCell = row.insertCell(1),
+              booleanCell = row.insertCell(2),
+              dateCell = row.insertCell(3),
+              deleteCell = row.insertCell(4),
+              rowNum = row.rowIndex - 1;
+  
+          tbl[rowNum] = [];
+  
+          addNameCell(nameCell);
+          addNumberCell(numberCell);
+          addBooleanCell(booleanCell);
+          addDateCell(dateCell);
+          addDeleteCell(deleteCell);
+  
+          assignEvents(row);
+  
+          tbl[rowNum].push('');
```

Comment:
LOL..
They are saved in the internal data structure Im using for sorting, The table contains only the values from the cells (not all html stuff)
Pushed it individually 'cause Im not good with JS programming yet.